### ISSUE TITLE
📝 Switch with fall through cases with comment gets fixed wrongly

### ISSUE THREAD

**Issue by anonrig** [biomejs/biome#1112]
### Environment information

```block
CLI:
  Version:                      1.4.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "alacritty"
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.3"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              true
  Organize imports disabled:    true
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

1. Go to https://github.com/getsentry/sentry
2. Clone it
3. Copy the following biome.json

```
{
  "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
  "vcs": {
    "enabled": true,
    "clientKind": "git",
    "useIgnoreFile": true
  },
  "organizeImports": {
    "enabled": false
  },
  "linter": {
    "enabled": false
  },
  "files": {
    "ignoreUnknown": true
  },
  "formatter": {
    "enabled": true,
    "formatWithErrors": true,
    "indentStyle": "space",
    "indentWidth": 2,
    "lineEnding": "lf",
    "ignore": [
      "api-docs",
      ".github",
      "tests",
      "fixtures",
      "scripts",
      "src",
      "static/app/data/world.json",
      "static/app/**/*trace.json",
      "package.json"
    ]
  },
  "javascript": {
    "formatter": {
      "enabled": true,
      "lineWidth": 90,
      "quoteStyle": "single",
      "jsxQuoteStyle": "double",
      "quoteProperties": "asNeeded",
      "trailingComma": "es5",
      "semicolons": "always",
      "arrowParentheses": "asNeeded",
      "bracketSpacing": false,
      "bracketSameLine": false
    }
  },
  "json": {
    "formatter": {
      "enabled": true
    }
  }
}
```

4. Run the following code:

```
npx @biomejs/biome check static/app/components/events/contexts/chunk.tsx
```

### Expected result

The proposed solution is wrong:

```
❯ npx @biomejs/biome check static/app/components/events/contexts/chunk.tsx
static/app/components/events/contexts/chunk.tsx format ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ Formatter would have printed the following content:

     46  46 │       case 'unity':
     47  47 │         return t('Unity');
     48     │ - ····case·'memory_info':·//·Future·new·value·for·memory·info
         48 │ + ····case·'memory_info':
         49 │ + ····//·Future·new·value·for·memory·info
     49  50 │       case 'Memory Info': // Current value for memory info
     50  51 │         return t('Memory Info');
     51     │ - ····case·'threadpool_info':·//·Future·new·value·for·thread·pool·info
         52 │ + ····case·'threadpool_info':
         53 │ + ····//·Future·new·value·for·thread·pool·info
     52  54 │       case 'ThreadPool Info': // Current value for thread pool info
     53  55 │         return t('Thread Pool Info');


static/app/components/events/contexts/chunk.tsx check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The file contains diagnostics that needs to be addressed.


Checked 1 file(s) in 943µs
Found 2 error(s)
check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.
```

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by faultyserver** [comment]
I believe this is fixed as part of #1035, which should be in the next release.

Specifically these lines of the diff are where it should be addressed: https://github.com/biomejs/biome/pull/1035/files#diff-96e1fbfdd959cd98a9b6ced692036cc2e9a6c7314a5243363cb3c25749b34179R74-R76

**Comment by Conaclos** [comment]
The issue is now fixed :)
